src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1464,6 +1481,71 @@ nns_edge_set_event_callback (nns_edge_h edge_h, nns_edge_event_cb cb,
   return NNS_EDGE_ERROR_NONE;
 }
 
+/**
+ * @brief Parse the message received from the MQTT broker and connect to the server directly.
+ */
+static int
+_mqtt_hybrid_direct_connection (nns_edge_handle_s * eh)
+{
+  int ret;
+
+  do {
+    char *msg = NULL;
+    char *server_ip = NULL;
+    int server_port = 0;
+    nns_size_t msg_len = 0;
+
+    ret = nns_edge_mqtt_get_message (eh->broker_h, (void **) &msg, &msg_len);
+    if (ret != NNS_EDGE_ERROR_NONE || !msg || msg_len == 0)
+      break;
+
+    nns_edge_parse_host_string (msg, &server_ip, &server_port);
+    SAFE_FREE (msg);
+
+    nns_edge_logd ("[DEBUG] Parsed server info: Server [%s:%d] ", server_ip,
+        server_port);
+
+    ret = _nns_edge_connect_to (eh, eh->client_id, server_ip, server_port);
+    SAFE_FREE (server_ip);
+
+    if (NNS_EDGE_ERROR_NONE == ret) {
+      break;
```

Comment:
How about 'return' instead of 'return' after 'break'?

Reply:
ditto